examples/vae/main.py

```diff
@@ -142,30 +139,37 @@ def eval(model, eval_ds, z):
 
 
 def main(argv):
-  key = random.PRNGKey(0)
+  del argv
+  rng = random.PRNGKey(0)
+  rng, key = random.split(rng)
+
   train_ds = tfds.load('mnist', split=tfds.Split.TRAIN)
   train_ds = train_ds.cache().shuffle(1000).batch(FLAGS.batch_size)
   test_ds = tfds.as_numpy(tfds.load('mnist', split=tfds.Split.TEST, batch_size=-1))
 
-  _, params = VAE.init_by_shape(key, [((1, 784), jnp.float32)])
+  _, params = VAE.init_by_shape(
+      key, [((1, 784), jnp.float32)], z_rng=random.PRNGKey(0))
   vae = nn.Model(VAE, params)
 
   optimizer = optim.Adam(learning_rate=FLAGS.learning_rate).create(vae)
 
+  rng, z_key, eval_rng = random.split(rng, 3)
+  z = random.normal(z_key, (64, 20))
+
   for epoch in range(FLAGS.num_epochs):
     for batch in tfds.as_numpy(train_ds):
+      rng, key = random.split(rng)
       batch['image'] = batch['image'].reshape(-1, 784) / 255.0
```

Comment:
After dividing by 255. the data becomes "continuous". You need to dynamically binarise it- both for train and eval. Many VAEs report results on dynamically binarised MNIST and on statically binarised MNIST (some fixed dataset, which I believe isn't a part of TFDS).

Reply:
Scrap half of the previous comment - TFDS *does* have the binarized mnist dataset. It's called `binarized_mnist` :)

Reply:
What is your recommendation here? To eval on both?

Reply:
Just stick to the `binarized_mnist` in training and eval. This should be equivalent to the "static MNIST" setup from VAE literature.

Reply:
done